cdap-runtime-ext-dataproc/src/main/java/io/cdap/cdap/runtime/spi/provisioner/existingdataproc/ExistingDataprocProvisioner.java

```diff
@@ -0,0 +1,295 @@
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package io.cdap.cdap.runtime.spi.provisioner.existingdataproc;
+
+import com.google.cloud.dataproc.v1.ClusterOperationMetadata;
+import com.google.common.annotations.VisibleForTesting;
+import com.google.common.base.Splitter;
+import io.cdap.cdap.runtime.spi.provisioner.Capabilities;
+import io.cdap.cdap.runtime.spi.provisioner.Cluster;
+import io.cdap.cdap.runtime.spi.provisioner.ClusterStatus;
+import io.cdap.cdap.runtime.spi.provisioner.PollingStrategies;
+import io.cdap.cdap.runtime.spi.provisioner.PollingStrategy;
+import io.cdap.cdap.runtime.spi.provisioner.Provisioner;
+import io.cdap.cdap.runtime.spi.provisioner.ProvisionerContext;
+import io.cdap.cdap.runtime.spi.provisioner.ProvisionerSpecification;
+import io.cdap.cdap.runtime.spi.provisioner.ProvisionerSystemContext;
+import io.cdap.cdap.runtime.spi.runtimejob.DataprocClusterInfo;
+import io.cdap.cdap.runtime.spi.runtimejob.DataprocRuntimeJobManager;
+import io.cdap.cdap.runtime.spi.runtimejob.RuntimeJobManager;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.io.IOException;
+import java.security.GeneralSecurityException;
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.HashSet;
+import java.util.Iterator;
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+import java.util.concurrent.TimeUnit;
+import java.util.regex.Pattern;
+
+/**
+ * Connects to existing Dataproc and execute Pipelines. This does not create or teardown Dataproc cluster.
+ */
+public class ExistingDataprocProvisioner implements Provisioner {
+
+  private static final Logger LOG = LoggerFactory.getLogger(ExistingDataprocProvisioner.class);
+  private static final ProvisionerSpecification SPEC = new ProvisionerSpecification(
+    "gcp-existing-dataproc", "Existing Dataproc",
+    "Connect and Execute jobs on existing Dataproc cluster.");
+  private static final String CLUSTER_PREFIX = "cdap-";
+  // Keys for looking up system properties
+  private static final String LABELS_PROPERTY = "labels";
+  private static final String BUCKET = "bucket";
+  private static final String RUNTIME_JOB_MANAGER = "runtime.job.manager";
+  // keys and values cannot be longer than 63 characters
+  // keys and values can only contain lowercase letters, numbers, underscores, and dashes
+  // keys must start with a lowercase letter
+  // keys cannot be empty
+  private static final Pattern LABEL_KEY_PATTERN = Pattern.compile("^[a-z][a-z0-9_-]{0,62}$");
```

Comment:
Duplicated in `ExistingDataprocConf`

Reply:
There are validations at multiple stages one for the user defined labels and the one one from the previous code base to access the system labels. 

I just reused the same names.